### DISCUSSION TITLE
Doc, Buttons and Switches vs SwitchText command and more

### DISCUSSION THREAD

**Discussion by BlissCA** [arendst/Tasmota#15989] [Q&A] [answered]
In the following doc:
[https://tasmota.github.io/docs/Buttons-and-Switches/#setoption114](url)
[https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode-15](url)

SwitchText command is not mentionned and I found out about this command by chance after many hours...
(However, StateText command is mentionned in this section)

Also, it would be great under SetOption114 or SwitchMode15 section , to tell that with the use of SetOption114 1 and SetOption(x) 15, and SwitchText, the Tasmota Integration in HA will automatically create a binary sensor based on SwitchText.

For Exemple:
SetOption114 1 (Detach switch from its corresponding relay)
SwitchMode1 15 (sends MQTT messages tele/%topic%/SENSOR ...)
SwitchText1 GarageDoorStatus

This will Publish tele/%topic%/SENSOR {"Time":"2022-07-15T23:32:26","GarageDoorStatus":"OFF"}
(Or "ON") everytime the switch changes state or every 5 minutes.

If you have for exemple a device "GarageDoor" in HA (switch.garagedoor), it will create automatically without the use of rules, the entity binary_sensor.garagedoorstatus attach to this device.

That would be nice to add in the doc, the switchtext command and HA binary_sensor auto creation when both SO1114 and SwitchMode15 are used...

Thank you.

Yves

**Comment by BlissCA** [comment]
I found a little mention of binary_sensor creation upon the use of setoption114 in the HA section.
[https://tasmota.github.io/docs/Home-Assistant/](url)

But it is not detailed as what SwitchMode does what when SO114 is 1.

**Comment by barbudor** [comment, marked as answer]
The docs, as the code, is open source so you are welcome to add that to the HA documentation page.

**Reply by BlissCA** [reply]
Thanks,

Unfortunately, I have no clues (yet) how to achieve that...  I'm just an end user :-)

Never managed to to figure out yet how this github thing is working...  forking, pull req etc...

However, I just noticed the dimmed pencil on the top right of a doc page...  I guess it starts from there ...  :-)

**Reply by BlissCA** [reply]
Ok so I just tried my first Fork/Pull Request...  :-)
Thanks...
